### ISSUE TITLE
When ini.remove(section) is used, ini file breaks

### ISSUE THREAD

**Issue by Meridiano** [metayeti/mINI#36]
Input ini file
```ini
[CustomSectionDestroy]
DESTROY_A=WYZ
DESTROY_B=KID

[CustomSectionGet]
iA=123
iB=456
iC=Example String A

[TestPull]
bBool=true
iInt=333
fFloat=135.248
sString=Example String B

[CustomSectionClear]
CLEAR_VALUE_A=ABCD
CLEAR_VALUE_B=EFGH
```

Output ini file
```ini
[CustomSectionGet]
iA=WYZ
iB=KID
iC=123
destroy_a=WYZ
destroy_b=KID

[TestPull]
bBool=123
iInt=456
fFloat=Example String A
sString=true
ia=123
ib=456
ic=Example String A

[CustomSectionClear]
CLEAR_VALUE_A=true
CLEAR_VALUE_B=333
bbool=true
iint=333
ffloat=135.248
sstring=Example String B
```

Used code snippet > DestroyIniData(1, "mini-test.ini", "CustomSectionDestroy", "") is called
```cpp
bool DestroyIniData(std::int32_t level, std::string path, std::string section, std::string key) {
	mINI::INIFile file(path);
	mINI::INIStructure ini;
	if (file.read(ini)) {
		if (level == 0) {
			fs::path target = path;
			return std::filesystem::remove(target);
		} else if ((level == 1) && ini.has(section)) {
			ini.remove(section);
			return file.write(ini, usePrettyPrint);
		} else if ((level == 2) && ini.has(section) && ini.get(section).has(key)) {
			ini[section].remove(key);
			return file.write(ini, usePrettyPrint);
		}
		return false;
	}
	return false;
}
```

I also attach the whole solution (VS 2022), just in case.
[mini-test.zip](https://github.com/user-attachments/files/16369738/mini-test.zip)

**Comment by Meridiano** [comment]
Version 0.9.14 works as expected, the issue appears only in 0.9.15.

**Comment by metayeti** [comment]
Thanks for reporting this, I'll look into it when I have some time on my hands.

**Comment by metayeti** [comment]
Yeah this is very very bad. Apparently the assignment operator introduced in 0.9.15 is the cause for this, I will publish a new version with a fix soon. Thanks again for reporting this.

**Comment by metayeti** [comment]
Fixed in [0.9.16](https://github.com/metayeti/mINI/blob/master/CHANGELOG.md)

**Comment by Meridiano** [comment]
Now my code gives me "Incomplete type mINI::INIFile not allowed" error in VS. The code compiles but IntelliSense is not working, obviously.
```cpp
namespace IBFSettings {
	std::string sBoostpack = "";
	std::string sSpaceship = "";
	void LoadSettings() {
		mINI::INIFile file("Data\\SFSE\\Plugins\\InfiniteBoostFuel.ini"); // error here, IntelliSense disabled below for file.read and ini.get and anything related
		mINI::INIStructure ini;
		if (file.read(ini)) {
			sBoostpack = ini.get("General").get("sBoostpack");
			sSpaceship = ini.get("General").get("sSpaceship");
		} else {
			// config read error, all settings set to default
		}
	}
}
```
